src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.c

```diff
@@ -211,180 +236,216 @@ void EnTanron5_Init(Actor* thisx, PlayState* play) {
-                        this->unk_19C *= 1.5f;
+                    f32 yFactor;
+                    f32 xzFactor;
+                    Vec3f fragmentAndDustPos;
+
+                    // Check the number of times this pyramid ruin has been hit to scale it accordingly.
+                    if (this->hitCount == 0) {
+                        yFactor = 180.0f;
+                        this->baseScale *= 1.4f;
+                    } else if (this->hitCount == 1) {
+                        yFactor = 230.0f;
+                        this->baseScale *= 1.37f;
+                    } else if (this->hitCount == 2) {
+                        yFactor = 780.0f;
+                        this->baseScale *= 1.5f;
                     }
-                    // TODO: determine if unk_1A0 ever has a different value from these 3, which will cause UB from spAC
-                    // being uninitialised
-                    this->actor.world.pos.y -= D_80BE5DD0 * spAC;
-                    Actor_SetScale(&this->actor, this->unk_19C * D_80BE5DD0);
-                    Math_Vec3f_Copy(&spB8, &this->actor.world.pos);
-
-                    for (i = 0; i < spC4; i++) {
-                        if (this->unk_1A0 == 0) {
-                            spA8 = 100.0f;
-                            spAC = 180.0f;
-                        } else if (this->unk_1A0 == 1) {
-                            spA8 = 200.0f;
-                            spAC = 100.0f;
-                        } else if (this->unk_1A0 == 2) {
-                            spA8 = 250.0f;
-                            spAC = 50.0f;
+
+                    this->actor.world.pos.y -= sGiantModeScaleFactor * yFactor;
+                    Actor_SetScale(&this->actor, this->baseScale * sGiantModeScaleFactor);
+                    Math_Vec3f_Copy(&pos, &this->actor.world.pos);
+
+                    for (i = 0; i < fragmentAndItemCount; i++) {
+                        if (this->hitCount == 0) {
+                            xzFactor = 100.0f;
+                            yFactor = 180.0f;
+                        } else if (this->hitCount == 1) {
+                            xzFactor = 200.0f;
+                            yFactor = 100.0f;
+                        } else if (this->hitCount == 2) {
+                            xzFactor = 250.0f;
+                            yFactor = 50.0f;
                         }
 
-                        sp9C.x = (Rand_CenteredFloat(spA8) * D_80BE5DD0) + spB8.x;
-                        sp9C.z = (Rand_CenteredFloat(spA8) * D_80BE5DD0) + spB8.z;
-                        sp9C.y = this->actor.floorHeight + (spAC * D_80BE5DD0);
+                        fragmentAndDustPos.x = (Rand_CenteredFloat(xzFactor) * sGiantModeScaleFactor) + pos.x;
+                        fragmentAndDustPos.z = (Rand_CenteredFloat(xzFactor) * sGiantModeScaleFactor) + pos.z;
+                        fragmentAndDustPos.y = this->actor.floorHeight + (yFactor * sGiantModeScaleFactor);
```

Comment:
```suggestion
                        fragmentAndDustPos.x = pos.x + (Rand_CenteredFloat(xzFactor) * sGiantModeScaleFactor);
                        fragmentAndDustPos.z = pos.z + (Rand_CenteredFloat(xzFactor) * sGiantModeScaleFactor);
                        fragmentAndDustPos.y = this->actor.floorHeight + (yFactor * sGiantModeScaleFactor);
```
I think this reads a little easier? (it matches)